libs/header/sync/sync_head.go

```diff
@@ -10,160 +10,131 @@ import (
 	"github.com/celestiaorg/celestia-node/libs/header"
 )
 
-// Head returns the Syncer's latest known header. It calls 'networkHead' in order to
-// either return or eagerly fetch the most recent header.
+// Head returns the Network Head.
+//
+// Known subjective head is considered network head if it is recent enough(now-timestamp<=blocktime)
+// Otherwise, head is requested from a trusted peer and
+// set as the new subjective head, assuming that trusted peer is always fully synced.
 func (s *Syncer[H]) Head(ctx context.Context) (H, error) {
-	return s.networkHead(ctx)
-}
-
-// subjectiveHead returns the latest known local header that is not expired(within trusting period).
-// If the header is expired, it is retrieved from a trusted peer without validation;
-// in other words, an automatic subjective initialization is performed.
-func (s *Syncer[H]) subjectiveHead(ctx context.Context) (H, error) {
```

Comment:
Don't be afraid of this red diff, it wasn't deleted. It was shortened and rearanged